app/models/audit/event/authn2.rb

```diff
@@ -0,0 +1,76 @@
+module Audit
+  module Event
+    class Authn
+
+      def initialize(
+          role:,
+          authenticator_name:,
+          service:,
+          success:,
+          operation:
+      )
+        @role = role
+        @authenticator_name = authenticator_name
+        @service = service
+        @success = success
+        @operation = operation
+      end
+
+      def progname
+        Event.progname
+      end
+
+      def severity
+        RubySeverity.new(possibly_failing_event.severity)
+      end
+
+      def authenticator_description
+        return @authenticator_name unless service_id
+        "#{@authenticator_name} service #{service_id}"
+      end
+
+      def service_id
+        @service&.resource_id
+      end
+
+      def message(success_msg:, failure_msg:, error_msg: nil)
+        possibly_failing_event.message(
+            success_msg: success_msg,
+            failure_msg: failure_msg,
+            error_msg: error_msg
+        )
+      end
+
+      def message_id
+        "authn"
+      end
+
+      def structured_data
+        {
+          SDID::SUBJECT => {role: @role&.id},
+          SDID::AUTH => auth_stuctured_data,
+        }.merge(
+          possibly_failing_event.action_sd(@operation)
```

Comment:
Is there a reason that `@operation` can't be passed to the constructor for `PossiblyFailingEvent`?

I'm not aware of any reason that would change in a given instance of an Audit event. So this could become just:
```
 possibly_failing_event.action_sd
```